crates/fuel-core/src/service/adapters/consensus_module/poa.rs

```diff
@@ -114,20 +116,27 @@ impl BlockImporter for BlockImporterAdapter {
 const TIME_UNTIL_SYNCED: Duration = Duration::from_secs(60);
 
 #[async_trait::async_trait]
-impl SyncPort for crate::service::adapters::SyncAdapter {
+impl SyncPort for crate::service::adapters::SyncAdapter<P2PAdapter> {
     async fn sync_with_peers(&mut self) -> anyhow::Result<()> {
-        // todo: connect to N amount of peers first!
-
-        'outer: loop {
-            tokio::select! {
-                _ = self.block_rx.recv() => {
-                    // keep receiving them blocks
-                }
-                _ = tokio::time::sleep(TIME_UNTIL_SYNCED) => {
-                    // time expired we are synced!
-                    break 'outer;
-                }
+        // todo: check if the next block to be produced equals last previous block + 1
+        // if yes, then we are synced already
+
+        // 1. check count of connected reserved peers
+        // todo: add 'n' tries or a timeout
+        loop {
+            let count = self.network_info.connected_reserved_peers().await;
```

Comment:
If there is a real error in the nested function this will loop forever

Reply:
As mentioned in the above comment - the only "error" is oneshot channel erroring, that should not be the case, and if it fails once, it should fail the next time

Reply:
If `self.network_info.connected_reserved_peers` silently fails inside and returns `0`, it will be a hot loop that will consume a lot of CPU. I agree with Tommi that it is better to return an error.

You work with an interface here, and implementation always can change in the future